cmd/attest.go

```diff
@@ -0,0 +1,218 @@
+				getInstanceMetadata()
+				addGceInstanceInfo(attestation)
+			}
+			if((len(gceProjectID) == 0 && gceProjectNumber != 0) || (len(gceProjectID) != 0 && gceProjectNumber == 0)){
+				return fmt.Errorf("attestation key type gceAK either use gceProjectId and gceProjectNumber together or none of them")
+			}
+
+		}else{
+			if(len(gceProjectID) != 0 || gceProjectNumber != 0){
+				return fmt.Errorf("attestation key type AK cannot use Gce Flags")
+			}
+		}
+		
+
+		if(len(outputText) != 0){
+			protoOut := []byte(prototext.Format(attestation))
+			if _,err := outputProto().Write(protoOut); err != nil{
+				return fmt.Errorf("failed to write attesattion report in textProto format: %v",err)
+			}
+		}
+
+		
+		if(len(output) != 0){
+			out ,err:= proto.Marshal(attestation)
+			if err != nil {
+				return fmt.Errorf("failed to marshal attestation proto: %v", attestation)
+			}
+			if _, err := dataOutput().Write(out); err != nil {
+					return fmt.Errorf("failed to write attestation proto: %v", err)
+			}
+		}
+		return nil
+	},
+	
+}
+
+func getInstanceMetadata() error{
+	if(len(gceZone) == 0){
+		out ,err := exec.Command("curl", "http://metadata.google.internal/computeMetadata/v1/instance/zone").Output()
+		if(err != nil){
+			return err
+		}
+		zone := string(out)
+		gceZone = zone[strings.Index(zone,"zones/")+6:]
+	}
+
+	if(gceInstanceID == 0){
+		id ,err := exec.Command("curl", "http://metadata.google.internal/computeMetadata/v1/instance/id").Output()
+		if(err != nil){
+			return err
+		}
+		gceInstanceID ,err = strconv.ParseUint(string(id),10,64)
+		if(err != nil){
+			return err
+		}
+	}
+
+	if(len(gceInstanceName)==0){
+		name ,err := exec.Command("curl", "http://metadata.google.internal/computeMetadata/v1/instance/name").Output()
+		if(err != nil){
+			return err
+		}
+		gceInstanceName = string(name)
+	}
+
+	return nil
+}
+func addGceInstanceInfo(attestation *attest.Attestation) {
+	attestation.InstanceInfo = &attest.GCEInstanceInfo{
+			Zone:          gceZone,
+			ProjectId:     gceProjectID,
+			ProjectNumber: gceProjectNumber,
+			InstanceName:  gceInstanceName,
+			InstanceId:    gceInstanceID,
+	}
+}
+func outputProto() io.Writer {
+	file, err := os.Create(outputText)
+	if err != nil {
+		return alwaysError{err}
+	}
+	return file
+}
+func addOutputTextFlag(cmd *cobra.Command){
+	cmd.PersistentFlags().StringVar(&outputText,"outputText","","output file where attestation report stored in text format")
+}
+func addKeyFlag(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&key, "key", "gceAK", "indicates type of attestation key to use <gceAK|AK>")
+}
+func addNonceflag(cmd *cobra.Command) {
+	cmd.PersistentFlags().BytesHexVar(&nonce, "nonce", []byte{}, "Hex encoded nonce for vTPM attestation , cannot be empty")
+}
+func addTeeNonceflag(cmd *cobra.Command) {
+	cmd.PersistentFlags().BytesHexVar(&teenonce, "teenonce", []byte{}, "Hex encoded teenonce for hardware attestation, can be empty")
```

Comment:
nit: stylistically, all comments should be lower-case. Also, remove unnecessary spaces: "hex encoded nonce for vTPM attestation, cannot be empty"

Reply:
Noted.
I'll change all the comments to lower-case.
Formatting issues can be resolved using gofmt.

Reply:
I'm sorry: I meant help text comments, not code comments. Comments should be capitalized and end with a period. Help text on flags should be lower case.

You still need to update the help text string here. gofmt does not format within strings to my knowledge.

Reply:
I have implemented it in the same way.
Also used gofmt with each modified file .
It is resolved from my side , please review it if required